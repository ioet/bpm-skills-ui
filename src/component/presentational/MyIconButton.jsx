import React from 'react';
import PropTypes from 'prop-types';
import IconButton from '@material-ui/core/IconButton/IconButton';
import Tooltip from '@material-ui/core/Tooltip/Tooltip';
import withStyles from '@material-ui/core/styles/withStyles';
import { TooltipConst } from '../../constants';
import { MyIconButtonStyles } from '../../styles';

const MyIconButton = (props) => {
  const {
    classes, categoryId, onClickCallback, icon, tooltip, hover, hoverId,
  } = props;
  const showItem = (hover && hoverId === categoryId) ? classes.show : classes.hide;

  return (
    <div>
      <Tooltip
        title={tooltip}
        placement="left"
        enterDelay={TooltipConst.ENTER_DELAY}
        leaveDelay={TooltipConst.LEAVE_DELAY}
      >
        <IconButton
          color="primary"
          onClick={(e) => {
            e.preventDefault();
            onClickCallback(categoryId);
          }}
          className={showItem}
        >
          {icon}
        </IconButton>
      </Tooltip>
    </div>
  );
};

MyIconButton.defaultProps = {
  hoverId: '',
};

MyIconButton.propTypes = {
  classes: PropTypes.object.isRequired,
  categoryId: PropTypes.string.isRequired,
  onClickCallback: PropTypes.func.isRequired,
  icon: PropTypes.element.isRequired,
  tooltip: PropTypes.string.isRequired,
  hover: PropTypes.bool.isRequired,
  hoverId: PropTypes.string,
};

export default withStyles(MyIconButtonStyles)(MyIconButton);
