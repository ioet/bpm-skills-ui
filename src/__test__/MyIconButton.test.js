import React from 'react';
import MyIconButton from '../component/presentational/MyIconButton';

describe('<MyIconButton />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<MyIconButton />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
