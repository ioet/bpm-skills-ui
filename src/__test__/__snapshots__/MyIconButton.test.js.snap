// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyIconButton /> Component Renders correctly 1`] = `
<MyIconButton
  classes={
    Object {
      "hide": "MyIconButton-hide-1",
      "show": "MyIconButton-show-2",
    }
  }
  hoverId=""
/>
`;
